kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/DeactivatedBeneficiaryServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package org.motechproject.nms.kilkari.service.impl;
+
+import org.motechproject.nms.kilkari.domain.DeactivatedBeneficiary;
+import org.motechproject.nms.kilkari.domain.DeactivationReason;
+import org.motechproject.nms.kilkari.repository.DeactivatedBeneficiaryDataService;
+import org.motechproject.nms.kilkari.service.DeactivatedBeneficiaryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service("deactivatedBeneficiaryService")
+public class DeactivatedBeneficiaryServiceImpl implements DeactivatedBeneficiaryService {
+
+    private DeactivatedBeneficiaryDataService deactivatedBeneficiaryDataService;
+
+    @Autowired
+    public DeactivatedBeneficiaryServiceImpl(DeactivatedBeneficiaryDataService deactivatedBeneficiaryDataService) {
+        this.deactivatedBeneficiaryDataService = deactivatedBeneficiaryDataService;
+    }
+
+    public List<DeactivatedBeneficiary> findByExternalId(String externalId) {
```

Comment:
Please change method name to reflect what it does -- this seems to only return some of the deactivatedbeneficiaries with that externalId